src/state/mod.rs

```diff
@@ -31,28 +32,108 @@ fn default_true() -> bool {
     true
 }
 
+#[derive(Debug, Clone, Default, serde::Serialize, serde::Deserialize)]
+pub struct ModGroup {
+    pub mods: Vec<ModConfig>,
+}
+
 #[obake::versioned]
 #[obake(version("0.0.0"))]
+#[obake(version("0.1.0"))]
 #[derive(Debug, Clone, Default, serde::Serialize, serde::Deserialize)]
 pub struct ModProfile {
     #[obake(cfg("0.0.0"))]
     pub mods: Vec<ModConfig>,
+
+    /// Mapping between mod groups and if they are enabled for a profile.
+    #[obake(cfg("0.1.0"))]
+    pub mod_groups: IndexMap<String, bool>,
```

Comment:
Is there a reason to use a map of booleans to indicate membership over an `IndexSet<String>`? It would look a lot cleaner when editing the JSON config by hand and would also avoid having to update membership of *every profile* when a group is added, removed, or renamed.

Reply:
Not really, I changed it to use `IndexSet<String>`.